torch/csrc/jit/passes/onnx/constant_map.cpp

```diff
@@ -0,0 +1,109 @@
+#include <torch/csrc/jit/passes/onnx/constant_map.h>
+
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+// Meyer’s Singleton for C++ 14
+ConstantValueMap& ConstantValueMap::getInstance() {
+  static ConstantValueMap s;
+  return s;
+}
+
+void ConstantValueMap::SetRank(
+    const std::string& tensorName,
+    size_t rankValue) {
+  ConstantValueMap::getInstance().rankMap.emplace(tensorName, rankValue);
+}
+
+bool ConstantValueMap::HasRank(const std::string& tensorName) {
+  return ConstantValueMap::getInstance().rankMap.find(tensorName) !=
+      ConstantValueMap::getInstance().rankMap.end();
+}
+
+c10::optional<size_t> ConstantValueMap::GetRank(const std::string& tensorName) {
+  if (!HasRank(tensorName)) {
+    return c10::nullopt;
+  }
+  return ConstantValueMap::getInstance().rankMap[tensorName];
+}
+
+void ConstantValueMap::SetShape(
+    const std::string& tensorName,
+    c10::VaryingShape<int64_t>& shapeValue) {
+  ConstantValueMap::getInstance().shapeMap.emplace(tensorName, shapeValue);
+}
+
+bool ConstantValueMap::HasShape(const std::string& tensorName) {
+  return ConstantValueMap::getInstance().shapeMap.find(tensorName) !=
+      ConstantValueMap::getInstance().shapeMap.end();
+}
+
+c10::optional<c10::VaryingShape<int64_t> > ConstantValueMap::GetShape(
+    const std::string& tensorName) {
+  if (!HasShape(tensorName)) {
+    return c10::nullopt;
+  }      
+  return ConstantValueMap::getInstance().shapeMap[tensorName];
+}
+
+void ConstantValueMap::SetValue(
+    const std::string& tensorName,
+    const at::Tensor& value) {
+  ConstantValueMap::getInstance().tensorValueMap.emplace(tensorName, value);
+}
+
+bool ConstantValueMap::HasValue(const std::string& tensorName) {
+  return ConstantValueMap::getInstance().tensorValueMap.find(tensorName) !=
+      ConstantValueMap::getInstance().tensorValueMap.end();
+}
+
+c10::optional<at::Tensor> ConstantValueMap::GetValue(const std::string& tensorName) {
+  if (!HasValue(tensorName)) {
+    return c10::nullopt;
+  }  
+  return ConstantValueMap::getInstance().tensorValueMap[tensorName];
+}
+
+void ConstantValueMap::ClearMaps() {
+  ConstantValueMap::getInstance().rankMap.clear();
+  ConstantValueMap::getInstance().shapeMap.clear();
+  ConstantValueMap::getInstance().tensorValueMap.clear();
+}
+
+// For debug only.
```

Comment:
if it's for debug only, do we need to add it to master?

Reply:
This is a useful debug tool, and we may have follow-up for this symbolic shape inference. Let's keep it for now.